tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarymindmap.code.tex

```diff
@@ -110,17 +101,41 @@
 }}%
 \tikzset{every circle connection bar/.style={}}%
 
+\pgfkeys{/pgf/decoration/start flare length/.initial=0pt}%
+\pgfkeys{/pgf/decoration/end flare length/.initial=0pt}%
```

Comment:
OK, I'm re-writing to use globals similarly to the the way \tikz@parse@switch does.  Please excuse my ignorance: is there any reason why I should not use a pair of \newdimen?  I ask because I see a lot of re-use of the same temporary variables, which makes me wonder if maybe variable allocations are a significant performance problem in this environment.

Reply:
To be clear, I do not intend for these to be customizable; I was just following the pattern of the code that marshals /pgf/decoration/start radius and end radius into the decoration macro.

Reply:
The avoidance of `\newdimen` dates back to the pre-eTeX times when there used to be only 256 dimen registers, so you wouldn't want to waste more than necessary.  In eTeX there are 32767 dimen registers, so running out is no longer a concern.

Reply:
Even if those parameters are not supposed to be modified, they are still part of the user interface and therefore require documentation.  See here:
https://github.com/pgf-tikz/pgf/blob/a9818b15a158f15a15cffcf7907c26f3f487005e/doc/generic/pgf/text-en/pgfmanual-en-library-decorations.tex#L80-L96